src/System.Net.WebSockets/tests/WebSocketExceptionTests.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+
+using Xunit;
+
+namespace System.Net.WebSockets.Tests
+{
+    public sealed class WebSocketExceptionTests
+    {
+        public static object[][] ErrorData = {
+            new object[] { WebSocketError.Success },
+            new object[] { WebSocketError.InvalidMessageType },
+            new object[] { WebSocketError.Faulted },
+            new object[] { WebSocketError.NativeError },
+            new object[] { WebSocketError.NotAWebSocket },
+            new object[] { WebSocketError.UnsupportedVersion },
+            new object[] { WebSocketError.UnsupportedProtocol },
+            new object[] { WebSocketError.HeaderError },
+            new object[] { WebSocketError.ConnectionClosedPrematurely },
+            new object[] { WebSocketError.InvalidState },
+        };
+
+        public static object[][] NativeErrorData = {
+            new object[] { 0, WebSocketError.Success },
+            new object[] { -2147467259, WebSocketError.NativeError },
+        };
+
+        public static object[][] UnrelatedErrorData =
+            ErrorData.SelectMany(wse => NativeErrorData.Select(ne => new object[] { wse[0], ne[0] })).ToArray();
+
+        [Theory, MemberData("ErrorData")]
```

Comment:
Cool... I didn't realize that MemberData supported fields like this... good to know.